pages/dashboard/index.js

```diff
@@ -0,0 +1,136 @@
+import React from 'react';
+import _ from 'lodash';
+import getConfig from 'next/config';
+import styled from 'styled-components';
+import { useTranslation } from 'react-i18next';
+
+import { useAuth } from '../../lib/Auth';
+import { useMatrix } from '../../lib/Matrix';
+import ServiceInvitations from './ServiceInvitations';
+import { ServiceTable } from '../../components/UI/ServiceTable';
+import DisplayInvitations from './DisplayInvitations';
+
+const sortRooms = function(room) {
+    return [
+        room.notificationCount === 0,
+        room.name,
+    ];
+};
+
+const TableSection = styled.section`
+  overflow-x: auto;
+
+  tbody tr:hover {
+    background-color: var(--color-background-alpha);
+  }
+`;
+export default function Dashboard() {
+    const auth = useAuth();
+    const matrix = useMatrix(auth.getAuthenticationProvider('matrix'));
+    const MatrixAuthProvider = auth.getAuthenticationProvider('matrix');
+    const { t } = useTranslation('dashboard');
+
+    const invites = matrix.invites;
+    const matrixClient = auth.getAuthenticationProvider('matrix').getMatrixClient();
```

Comment:
I think this is getting a little confusing overall... we have too many layers:

1. `useMatrix()` -- which is referring to things & functions inside `Matrix.js`
2. `auth.getAuthenticationProvider('matrix')` -- referring to functions inside `MatrixAuthProvider.js`
3. `auth.getAuthenticationProvider('matrix').getMatrixClient()` -- referring to the actual "raw" `matrix-js-sdk` `MatrixClient`

Do you have an idea how to simplify this?
One thing I can think of would be to never call (2) from anywhere, except for from `Matrix.js`. Which would mean that all functions we've added in `MatrixAuthProvider.js` (like `addSpaceChild`) would live in `Matrix.js` instead.

Reply:
maybe i'll change this in a separate PR? the same would go for `removeSpaceChild`